build-tools/plots-to-appinsights/Program.cs

```diff
@@ -0,0 +1,342 @@
+			}
+
+			return eventName;
+		}
+
+		static Dictionary<string, string> GetBaseTelemetryProperties (Settings settings)
+		{
+			if (settings == null) {
+				throw new ArgumentNullException (nameof (settings));
+			}
+
+			var schemaVersion = "1.0";
+			var sessionTimestamp = DateTime.UtcNow.ToString (Constants.AzureTimestampFormatUtc);
+
+			var properties = new Dictionary<string, string> (StringComparer.OrdinalIgnoreCase) {
+					{ "Version", schemaVersion },
+					{ "Environment", settings.Environment },
+					{ "SessionTimestamp", sessionTimestamp },
+					{ "BuildSystem", settings.BuildSystem },
+					{ "BuildPipeline", settings.BuildPipelineName },
+					{ "BuildType", settings.BuildType },
+					{ "BuildId", settings.BuildId },
+					{ "BuildNumber", settings.BuildNumber },
+					{ "BuildUrl", settings.BuildUrl },
+					{ "BuildDate", settings.BuildDateUtc.ToString(Constants.AzureTimestampFormatUtc) },
+					{ "Filename", Path.GetFileName(settings.CsvPathAndFilename) },
+					{ "Title", settings.PlotTitle },
+					{ "Group", settings.PlotGroup }
+				};
+
+			return properties;
+		}
+
+		static void SendTelemtry (AppInsights appInsightsClient, string eventName, Dictionary<string, decimal> plots, Settings settings, TextWriter tw)
+		{
+			if (appInsightsClient == null) {
+				throw new ArgumentNullException (nameof (appInsightsClient));
+			}
+
+			if (string.IsNullOrEmpty (eventName)) {
+				throw new ArgumentNullException (nameof (eventName));
+			}
+
+			if (plots == null || !plots.Any()) {
+				throw new ArgumentNullException (nameof (plots));
+			}
+
+			if (settings == null) {
+				throw new ArgumentNullException (nameof (settings));
+			}
+
+			if (tw == null) {
+				throw new ArgumentNullException (nameof (tw));
+			}
+
+			var ctPlot = 0;
+			foreach (var plot in plots) {
+
+				var properties = GetBaseTelemetryProperties (settings);
+				var plotProperties = new Dictionary<string, string> (StringComparer.OrdinalIgnoreCase) {
+					{ "SeriesLabel", plot.Key }
+				};
+
+				// Add plot information to the properties dictionary
+				plotProperties.ToList ().ForEach (x => properties [x.Key] = x.Value);
+
+				var metrics = new Dictionary<string, double> (StringComparer.OrdinalIgnoreCase) {
+					{ "Value", (double) plot.Value }
+				};
+
+				appInsightsClient.SendTelemetry (eventName, properties, metrics);
+
+				ctPlot++;
+				tw.WriteLine ($"Telemetry sent: Plot {ctPlot}");
+				tw.WriteLine ($"  Event name: {eventName}");
+				tw.WriteLine ("    Properties:");
+				foreach (var property in properties.OrderBy (x => x.Key)) {
+					tw.WriteLine ($"      {property.Key}: {property.Value}");
+				}
+
+				tw.WriteLine ("    Metrics:");
+				foreach (var metric in metrics.OrderBy (x => x.Key)) {
+					tw.WriteLine ($"      {metric.Key}: {metric.Value}");
+				}
+			}
+		}
+
+		static void SendTelemetry_Warnings (AppInsights appInsightsClient, string eventName, List<string> warnings, Settings settings, TextWriter tw)
```

Comment:
Method names shouldn't have `_` except in unit tests.

Reply:
Nuked the underscores. Commit: 56a0b717242a0d9ad9665993b7bfb9dff258eb2d